### PR TITLE
feat: integrate banners into searchResponse

### PR DESCRIPTION
**Pull request by chidojiro**

### CODE REVIEW

**Comment by zlatanpham** [comment]
@chidojiro bare with me while I'm updating the Console to be able to test with the real data. I will be back to review the PR soon.

File: src/index.ts [hunk 1]
```diff
@@ -498,6 +498,9 @@ class QueryPipeline extends EventEmitter {
           jsonProto.searchResponse?.totalResults || "0",
           10
         ),
+        ...(jsonProto.searchResponse?.banners && {
+          banners: jsonProto.searchResponse?.banners,
+        }),
```
Comment (zlatanpham) [inline src/index.ts:503]: `banners` should be referred from the outer level. Run the CURL and you can see it in the returned response:

```
curl 'http://jsonapi-au-staging-valkyrie2.sajari.com/sajari.api.pipeline.v1.Query/Search' \
  -H 'Connection: keep-alive' \
  -H 'Accept: application/json' \
  -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/97.0.4692.99 Safari/537.36' \
  -H 'Content-Type: text/plain' \
  -H 'Origin: http://localhost:4000' \
  -H 'Referer: http://localhost:4000/' \
  -H 'Accept-Language: en-US,en;q=0.9,de;q=0.8,fi;q=0.7,it;q=0.6,pt;q=0.5,ru;q=0.4,vi;q=0.3,fr;q=0.2,ja;q=0.1,sv;q=0.1,ko;q=0.1,cs;q=0.1,es;q=0.1,da;q=0.1,id;q=0.1,is;q=0.1,nl;q=0.1,la;q=0.1,zh-CN;q=0.1,zh;q=0.1' \
  --data-raw $'{"metadata":{"project":["1617855295958267636"],"collection":["sajari-thanh-pham"],"user-agent":["sajari-sdk-js/2.8.2 (via localhost:4000)"]},"request":{"pipeline":{"name":"query"},"tracking":{"query_id":"ejaf27qxlbsf2uyf","type":"NONE","sequence":12,"field":"_id","data":{}},"values":{"q":"black","resultsPerPage":"15","fields":"","filter":"_id \u0021= \\"\\"","countFilters":"","buckets":"","count":"","min":"","max":""}}}' \
  --compressed \
  --insecure
```

File: src/index.ts [hunk 2]
```diff
@@ -498,6 +498,9 @@ class QueryPipeline extends EventEmitter {
           jsonProto.searchResponse?.totalResults || "0",
           10
         ),
+        ...(jsonProto.banners && {
+          banners: jsonProto.banners,
+        }),
```
Comment (zlatanpham) [inline src/index.ts:503]: Can it be simplified like this?

```suggestion
        banners: jsonProto?.banners,
```

**Review by zlatanpham — APPROVED** [review]
One nit, otherwise, LGTM.

**Review by zlatanpham — APPROVED** [review]

File: src/index.ts [hunk 3]
```diff
@@ -552,6 +559,90 @@ export interface SearchResponse {
...
+  BottomRight = "TEXT_POSITION_BOTTOM_RIGHT",
+}
+
+export interface Banner {
+  /**
+   * The description of the banner, displayed in sub-head font.
+   * @type {string}
+   * @memberof Banner
+   */
+  description?: string;
```
Comment (wwalser) [inline src/index.ts:589]: Are all of these fields actually optional?
It seems to me that the banner itself may be optional but if a banner is present, many of these fields are required. No?

Comment (zlatanpham) [reply src/index.ts:589]: We took the interface from the generated code. However, it's true that only text fields are optional, the rest are required to create a banner. Do you think we should make an update to reflect that?

Comment (wwalser) [reply src/index.ts:589]: Fair enough. Let's leave them as optional.

**Review by wwalser — APPROVED** [review]

**Pull request merged**